src/dist/replication/lib/replica_config.cpp

```diff
@@ -772,6 +773,7 @@ bool replica::update_local_configuration(const replica_configuration &config,
             break;
         case partition_status::PS_INACTIVE:
             _primary_states.cleanup(old_ballot != config.ballot);
+            _bulk_loader->clear_bulk_load_states();
```

Comment:
If possible, reduce the `_bulk_loader`-related code in class `replica`, which is bloated as more functions added.

A feasible way is:

```
    _bulk_loader->clear_bulk_load_states_if_needed(); // Add here.

    switch (old_status) {
    case partition_status::PS_PRIMARY:
        cleanup_preparing_mutations(false);
        switch (config.status) {
        case partition_status::PS_PRIMARY:
            replay_prepare_list();
            break;
        case partition_status::PS_INACTIVE:
            _primary_states.cleanup(old_ballot != config.ballot);
```

```cpp
void replica_bulk_loader::clear_bulk_load_states_if_needed(old_status) {
    switch (old_status) {
    case partition_status::PS_PRIMARY:
        switch (config.status) {
        case partition_status::PS_INACTIVE:
        case partition_status::PS_ERROR:
            clear_bulk_load_states();
            break;
    case partition_status::PS_SECONDARY:
            clear_bulk_load_states();
            break;         
    }
}
```